File: api_examples/apiget.py
#!/usr/bin/env python2.7
#
# demo python cloudpassage API stuff
# Tim Spencer <tspencer@cloudpassage.com>
#
# This should be a lot simpler than it is, but none of the oath2
# modules seem to work well, so we do it by hand here.
#
clientid = 'XXXXXXXX'
clientsecret = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX' 
host = 'api.cloudpassage.com'


import urllib
import httplib
import base64
import json

# Get the access token used for the API calls.
connection = httplib.HTTPSConnection(host)
authstring = "Basic " + base64.b64encode(clientid + ":" + clientsecret)
header = {"Authorization": authstring}
params = urllib.urlencode({'grant_type': 'client_credentials'})
connection.request("POST", '/oauth/access_token', params, header)
response = connection.getresponse()
jsondata =  response.read().decode()
data = json.loads(jsondata)
key = data['access_token']

# Do the real request using the access token in the headers
tokenheader = {"Authorization": 'Bearer ' + key}
connection.request("GET", "/v1/servers", '', tokenheader)
response = connection.getresponse()
jsondata =  response.read().decode()
data = json.loads(jsondata)

# print out everything in a pretty way
#print json.dumps(data, sort_keys=True, indent=4)

# iterate through the list and print out the hostnames as an example of
# how to handle json data
servers = data['servers']
for server in servers:
	print server['hostname']

connection.close()

File: api_examples/apiget.rb
#!/usr/bin/env ruby
#
# demo ruby cloudpassage API stuff
# Tim Spencer <tspencer@cloudpassage.com>
#
# you may need to install the oauth2 and rest-client gems
#
clientid = 'XXXXXXXX'
clientsecret = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX' 
host = 'api.cloudpassage.com'


require 'oauth2'
require 'rest-client'
require 'json'

#To accomodate a proxy, we need to handle both RestClient with the
#following one-time statement, and also as a :proxy parameter to the
#oauth2 call below.
my_proxy = nil
if ENV['https_proxy'].to_s.length > 0
  my_proxy = ENV['https_proxy']
  RestClient.proxy = my_proxy
  $stderr.puts "Using proxy: #{RestClient.proxy}"
end

client = OAuth2::Client.new(clientid, clientsecret,
        :connection_opts => { :proxy => my_proxy },
        :site => "https://#{host}",
        :token_url => '/oauth/access_token'
)
token = client.client_credentials.get_token.token

result = RestClient.get "https://#{host}/v1/servers", {
        'Authorization' => "Bearer #{token}"
}

data = JSON result.body
servers = data['servers']
servers.each do |server|
	puts server['connecting_ip_address'] + " " + server['hostname']
	#puts server['hostname']
end

